File: lt0172.cpp
class Solution {
public:
    int trailingZeroes(int n) {
        int five_num = 0;
        for(long long i = 5; i <= n; i = i * 5)
        {
            five_num += n / i;
        }
        
        return five_num;
    }
};

File: code/lt0172.java
// https://leetcode-cn.com/problems/factorial-trailing-zeroes/
// factorial trailing zeroes
class Solution {
    public int trailingZeroes(int n) {
        int count = 0;
        while(n / 5 != 0){
            count += n / 5;
            n = n / 5;
        }
        return count;
    }
}